android/titanium/src/java/org/appcelerator/titanium/util/TiOrientationHelper.java

```diff
@@ -18,54 +17,38 @@
 	public static final int ORIENTATION_LANDSCAPE = 2;
 	public static final int ORIENTATION_PORTRAIT_REVERSE = 3;
 	public static final int ORIENTATION_LANDSCAPE_REVERSE = 4;
-	public static final int ORIENTATION_SQUARE = 5;
+
+	public static int orientation = ORIENTATION_UNKNOWN;
+	public static boolean isPortrait = true;
 
 	public static int convertRotationToTiOrientationMode (int rotation, int width, int height)
 	{
-	    // Device that has portrait natural orientation
-	    if ((rotation == Surface.ROTATION_0 || rotation == Surface.ROTATION_180) && height > width ||
-	            (rotation == Surface.ROTATION_90 || rotation == Surface.ROTATION_270) && width > height) {
-
 	        switch (rotation)
 	        {
 	            case Surface.ROTATION_0:
+					isPortrait = true;
+	            	orientation = ORIENTATION_PORTRAIT;
 	                return ORIENTATION_PORTRAIT;
 
 	            case Surface.ROTATION_90:
+					isPortrait = false;
+					orientation = ORIENTATION_LANDSCAPE;
 	                return ORIENTATION_LANDSCAPE;
 
 	            case Surface.ROTATION_180:
+					isPortrait = true;
+					orientation = ORIENTATION_PORTRAIT_REVERSE;
 	                return ORIENTATION_PORTRAIT_REVERSE;
 
 	            case Surface.ROTATION_270:
+					isPortrait = false;
+					orientation = ORIENTATION_LANDSCAPE_REVERSE;
 	                return ORIENTATION_LANDSCAPE_REVERSE;
 
 	            default:
+	            	orientation = ORIENTATION_UNKNOWN;
 	                return ORIENTATION_UNKNOWN;
 	        }
-
-	        // Device that has landscape natural orientation. Eg Samsung Galaxy Tab 3 10.1
-	    } else {
-	        switch (rotation)
-	        {
-	            case Surface.ROTATION_0:
-	                return ORIENTATION_LANDSCAPE;
-
-	            case Surface.ROTATION_90:
-	                return ORIENTATION_PORTRAIT;
-
-	            case Surface.ROTATION_180:
-	                return ORIENTATION_LANDSCAPE_REVERSE;
-
-	            case Surface.ROTATION_270:
-	                return ORIENTATION_PORTRAIT_REVERSE;
-
-	            default:
-	                return ORIENTATION_UNKNOWN;
-	        }
-
-	    }
-
```

Comment:
Also note that I call the new method `getWindowTiOrientationModeFrom()` to specify that it's fetching the orientation of the application window. In 7.0.0, we'll look into fetching device orientation, but it may still be useful to know the window orientation as well for our own internal purposes.